From 72aeb4c0bf1ae130c0c40a0f08e8dd34d14c4e4c Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 21 Feb 2024 17:16:42 +0100
Subject: [PATCH 0657/1695] dts: stm32mp13: add operating point nodes and
 configuration

Add the OPP table configuration for the STM32MP131 platform
and its associated configuration depending to SoC version.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 55616c4dbf66d6a3d86a03788a0f05fa9f4460f0 dts: stm32mp13: add operating point nodes and configuration]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: Ia6ce1ba2a3d137f18843dbc81e9ec3959fb93f3e
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/279936
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362295
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi  | 23 +++++++++++++++++++++++
 core/arch/arm/dts/stm32mp13xa.dtsi |  5 +++++
 core/arch/arm/dts/stm32mp13xc.dtsi |  1 +
 core/arch/arm/dts/stm32mp13xd.dtsi |  5 +++++
 core/arch/arm/dts/stm32mp13xf.dtsi |  1 +
 5 files changed, 35 insertions(+)
 create mode 100644 core/arch/arm/dts/stm32mp13xa.dtsi
 create mode 100644 core/arch/arm/dts/stm32mp13xd.dtsi

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index b38df95ac..c21439a28 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -22,6 +22,29 @@
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
 			reg = <0>;
+			clocks = <&rcc CK_MPU>;
+			clock-names = "cpu";
+			operating-points-v2 = <&cpu0_opp_table>;
+			nvmem-cells = <&part_number_otp>;
+			nvmem-cell-names = "part_number";
+		};
+	};
+
+	cpu0_opp_table: cpu0-opp-table {
+		compatible = "operating-points-v2";
+
+		opp-650000000 {
+			opp-hz = /bits/ 64 <650000000>;
+			opp-microvolt = <1250000>;
+			opp-supported-hw = <0x3>;
+			st,opp-default;
+		};
+
+		opp-1000000000 {
+			opp-hz = /bits/ 64 <1000000000>;
+			opp-microvolt = <1350000>;
+			opp-supported-hw = <0x2>;
+			st,opp-default;
 		};
 	};
 
diff --git a/core/arch/arm/dts/stm32mp13xa.dtsi b/core/arch/arm/dts/stm32mp13xa.dtsi
new file mode 100644
index 000000000..5fb0294e6
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp13xa.dtsi
@@ -0,0 +1,5 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
diff --git a/core/arch/arm/dts/stm32mp13xc.dtsi b/core/arch/arm/dts/stm32mp13xc.dtsi
index 4527f1105..51bab0c0d 100644
--- a/core/arch/arm/dts/stm32mp13xc.dtsi
+++ b/core/arch/arm/dts/stm32mp13xc.dtsi
@@ -3,6 +3,7 @@
  * Copyright (C) STMicroelectronics 2021-2022 - All Rights Reserved
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
+#include "stm32mp13xa.dtsi"
 
 &etzpc {
 	cryp: crypto@54002000 {
diff --git a/core/arch/arm/dts/stm32mp13xd.dtsi b/core/arch/arm/dts/stm32mp13xd.dtsi
new file mode 100644
index 000000000..5fb0294e6
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp13xd.dtsi
@@ -0,0 +1,5 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
diff --git a/core/arch/arm/dts/stm32mp13xf.dtsi b/core/arch/arm/dts/stm32mp13xf.dtsi
index 4527f1105..dd106b50f 100644
--- a/core/arch/arm/dts/stm32mp13xf.dtsi
+++ b/core/arch/arm/dts/stm32mp13xf.dtsi
@@ -3,6 +3,7 @@
  * Copyright (C) STMicroelectronics 2021-2022 - All Rights Reserved
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
+#include "stm32mp13xd.dtsi"
 
 &etzpc {
 	cryp: crypto@54002000 {
-- 
2.47.2

